pkg/ingest/conn.go

```diff
@@ -55,76 +57,128 @@ func HandleConnections(
 		// In either case, the writer is closed.
 		m.register(conn)
 		go func() {
-			h(conn, w, idGen, connectedClients)
+			defer conn.Close()
+			h(reader(conn), w, idGen, connectedClients)
 			w.Stop() // make sure it's flushed
 			m.remove(conn)
 		}()
 	}
 }
 
+// ErrIllegalTopicFormat is returned if a record did not start with a proper topic prefix.
+var ErrIllegalTopicFormat = errors.New("illegal topic formatting")
+
+// RecordReader emits records that are prefixed with a topic.
+// It returns io.EOF if the underlying record source has no more data.
+type RecordReader func() (record []byte, err error)
```

Comment:
I like this even independent of the topic feature that motivates it, because if we can replace all bufio.Readers with RecordReader universally, it will give me a way to elegantly solve #21.

I guess it's out of scope for this PR, but what do you think about it in general?

Reply:
Yes, I think that's a good way to go. Ideas for a shared package name where this could go into?

Reply:
I think `pkg/record` is totally appropriate, since OK Log settles on "log record" as the domain nomenclature.